src/Storages/Distributed/DirectoryMonitor.cpp

```diff
@@ -202,7 +208,7 @@ namespace
 
     void writeRemoteConvert(const DistributedHeader & header, RemoteBlockOutputStream & remote, ReadBufferFromFile & in, Poco::Logger * log)
     {
-        if (remote.getHeader() && header.header != remote.getHeader().dumpStructure())
+        if (remote.getHeader() && header.names_and_types != remote.getHeader().dumpNamesAndTypes())
```

Comment:
Does sending and comparing string representations of block's names and types really gives gain instead of reading one block?
Actually `dumpNamesAndTypes` also looks like debug method like `dumpStructure`. 

Maybe we can send header on native format?